src/frontend/utils/chat/chat.ts

```diff
@@ -0,0 +1,31 @@
+import { converse } from 'utils/window';
+
+export const generateAnonymousNickname = (): string => {
+  const tokenLength = 8;
+  const charUniverse =
+    'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890'.split('');
+  const generatedToken = [];
+  for (let i = 0; i < tokenLength; i++) {
+    const j = Math.round(Math.random() * (charUniverse.length - 1));
+    generatedToken[i] = charUniverse[j];
+  }
+  return 'Anonymous-'.concat(generatedToken.join(''));
```

Comment:
if you stick to this generation you can factorise the two methods here

Reply:
As said  in the previous comment, I will delete the generateAnonymousToken, so there is no need to factorise anymore here.